server/src/main/java/org/elasticsearch/common/settings/IndexScopedSettings.java

```diff
@@ -51,180 +54,192 @@
-        IndexSettings.INDEX_TRANSLOG_FLUSH_THRESHOLD_AGE_SETTING,
-        IndexSettings.INDEX_TRANSLOG_GENERATION_THRESHOLD_SIZE_SETTING,
-        IndexSettings.INDEX_TRANSLOG_RETENTION_AGE_SETTING,
-        IndexSettings.INDEX_TRANSLOG_RETENTION_SIZE_SETTING,
-        IndexSettings.INDEX_SEARCH_IDLE_AFTER,
-        IndexSettings.INDEX_SEARCH_THROTTLED,
-        IndexFieldDataService.INDEX_FIELDDATA_CACHE_KEY,
-        IndexSettings.IGNORE_ABOVE_SETTING,
-        FieldMapper.IGNORE_MALFORMED_SETTING,
-        FieldMapper.COERCE_SETTING,
-        Store.INDEX_STORE_STATS_REFRESH_INTERVAL_SETTING,
-        MapperService.INDEX_MAPPING_NESTED_FIELDS_LIMIT_SETTING,
-        MapperService.INDEX_MAPPING_NESTED_DOCS_LIMIT_SETTING,
-        MapperService.INDEX_MAPPING_IGNORE_DYNAMIC_BEYOND_LIMIT_SETTING,
-        MapperService.INDEX_MAPPING_TOTAL_FIELDS_LIMIT_SETTING,
-        MapperService.INDEX_MAPPING_DEPTH_LIMIT_SETTING,
-        MapperService.INDEX_MAPPING_DIMENSION_FIELDS_LIMIT_SETTING,
-        MapperService.INDEX_MAPPING_FIELD_NAME_LENGTH_LIMIT_SETTING,
-        MapperService.INDEX_MAPPER_DYNAMIC_SETTING,
-        BitsetFilterCache.INDEX_LOAD_RANDOM_ACCESS_FILTERS_EAGERLY_SETTING,
-        IndexModule.INDEX_STORE_TYPE_SETTING,
-        IndexModule.INDEX_STORE_PRE_LOAD_SETTING,
-        IndexModule.INDEX_RECOVERY_TYPE_SETTING,
-        IndexModule.INDEX_QUERY_CACHE_ENABLED_SETTING,
-        FsDirectoryFactory.INDEX_LOCK_FACTOR_SETTING,
-        EngineConfig.INDEX_CODEC_SETTING,
-        IndexMetadata.SETTING_WAIT_FOR_ACTIVE_SHARDS,
-        IndexSettings.DEFAULT_PIPELINE,
-        IndexSettings.FINAL_PIPELINE,
-        MetadataIndexStateService.VERIFIED_BEFORE_CLOSE_SETTING,
-        MetadataIndexStateService.VERIFIED_READ_ONLY_SETTING,
-        ExistingShardsAllocator.EXISTING_SHARDS_ALLOCATOR_SETTING,
-        DiskThresholdDecider.SETTING_IGNORE_DISK_WATERMARKS,
-        ShardLimitValidator.INDEX_SETTING_SHARD_LIMIT_GROUP,
-        DataTier.TIER_PREFERENCE_SETTING,
-        IndexSettings.BLOOM_FILTER_ID_FIELD_ENABLED_SETTING,
-        IndexSettings.LIFECYCLE_ORIGINATION_DATE_SETTING,
-        IndexSettings.LIFECYCLE_PARSE_ORIGINATION_DATE_SETTING,
-        IndexSettings.TIME_SERIES_ES87TSDB_CODEC_ENABLED_SETTING,
-        IndexSettings.LOGSDB_ROUTE_ON_SORT_FIELDS,
-        IndexSettings.LOGSDB_SORT_ON_HOST_NAME,
-        IndexSettings.LOGSDB_ADD_HOST_NAME_FIELD,
-        IndexSettings.PREFER_ILM_SETTING,
-        DataStreamFailureStoreDefinition.FAILURE_STORE_DEFINITION_VERSION_SETTING,
-        FieldMapper.SYNTHETIC_SOURCE_KEEP_INDEX_SETTING,
-        IgnoredSourceFieldMapper.SKIP_IGNORED_SOURCE_WRITE_SETTING,
-        IgnoredSourceFieldMapper.SKIP_IGNORED_SOURCE_READ_SETTING,
-        IndexSettings.INDEX_MAPPER_SOURCE_MODE_SETTING,
-        IndexSettings.RECOVERY_USE_SYNTHETIC_SOURCE_SETTING,
-        InferenceMetadataFieldsMapper.USE_LEGACY_SEMANTIC_TEXT_FORMAT,
+    public static final Set<Setting<?>> BUILT_IN_INDEX_SETTINGS;
```

Comment:
This change will not be here once we merge https://github.com/elastic/elasticsearch/pull/121751